src/query/policy/resolver/interface.go

```diff
@@ -36,6 +36,7 @@ type PolicyResolver interface {
 	// may still be no data retained for the range in the given storage policy.
 	Resolve(
 		ctx context.Context,
+		tagOptions models.TagOptions,
```

Comment:
nit: Perhaps move to last arg since it's an options arg and no others are?

Reply:
sgtm